components/Form/DateInput/index.js

```diff
@@ -124,13 +129,18 @@ const DateInput = ({
                 id={`${name}-year`}
                 name={`${name}-year`}
                 type="text"
-                pattern="[0-9]*"
+                pattern="^[0-9]{4}$"
+                required={required}
                 inputMode="numeric"
                 onChange={e => {
-                  setYearValue(e.target.value);
                   onChange(e);
                 }}
-                value={yearValue}
+                onInvalid={e => {
```

Comment:
Will this fire as soon as you start typing numbers? ie as soon as someone hits '2'?

Reply:
No, it will fire only after you submit the form

Reply:
great!